python/mit/binding.py.in

```diff
@@ -62,20 +63,47 @@ sign_bit = 1 << (word_bit - 1)
 
 
 # Types
-if word_bytes == sizeof(c_short):
-    c_word = c_short
-    c_uword = c_ushort
-elif word_bytes == sizeof(c_int):
-    c_word = c_int
-    c_uword = c_uint
-elif word_bytes == sizeof(c_long):
-    c_word = c_long
-    c_uword = c_ulong
-elif word_bytes == sizeof(c_longlong):
-    c_word = c_longlong
-    c_uword = c_ulonglong
+class c_int128(Structure):
+    _fields_ = [('low', c_uint64),
+                ('high', c_int64)]
+
+    def __init__(self, n=0):
+        self.low = n & ((1 << 64) - 1)
+        self.high = n >> 64
+
+    @classmethod
+    def from_param(cls, value):
+        return cls(value)
+
+    @property
+    def value(self):
+        return self.low | (self.high << 64)
+
+    def __int__(self):
+        return self.value
+
+    def __hash__(self):
```

Comment:
If you're using `__hash__` you probably need `__eq__` too.

Reply:
Done.